### PR TITLE
Add more TimeSpan formatting tests

### PR DESCRIPTION
**Pull request by stephentoub**
cc: @danmosemsft

### CODE REVIEW

File: src/System.Runtime/tests/System/TimeSpanTests.cs [hunk 1]
```diff
@@ -848,48 +848,135 @@ public static void Subtract_Invalid()
...
+            yield return new object[] { input, "fffffff", invariantInfo, "9101112" };
+            yield return new object[] { input, "%F", invariantInfo, "9" };
+            yield return new object[] { input, "FF", invariantInfo, "91" };
+            yield return new object[] { input, "FFF", invariantInfo, "91" };
+            yield return new object[] { input, "FFFF", invariantInfo, "9101" };
+            yield return new object[] { input, "FFFFF", invariantInfo, "91011" };
+            yield return new object[] { input, "FFFFFF", invariantInfo, "910111" };
+            yield return new object[] { input, "FFFFFFF", invariantInfo, "9101112" };
+            yield return new object[] { input, "dd\\.ss", invariantInfo, "142.18" };
+            yield return new object[] { input, "dd\\.ss", commaSeparatorInfo, "142.18" };
```
Comment (danmosemsft) [inline src/System.Runtime/tests/System/TimeSpanTests.cs]: Maybe more `d`'s so as to get leading zeros?

File: src/System.Runtime/tests/System/TimeSpanTests.cs [hunk 2]
```diff
@@ -848,48 +848,135 @@ public static void Subtract_Invalid()
...
+            // constant/invariant format
+            foreach (CultureInfo info in new[] { null, invariantInfo, commaSeparatorInfo }) // validate that culture is ignored
+            {
+                foreach (string constFormat in new[] { null, "c", "t", "T" })
+                {
+                    yield return new object[] { input, constFormat, info, "142.21:21:18.9101112" };
+                    yield return new object[] { new TimeSpan(0), constFormat, info, "00:00:00" };
+                    yield return new object[] { new TimeSpan(1), constFormat, info, "00:00:00.0000001" };
+                    yield return new object[] { new TimeSpan(-1), constFormat, info, "-00:00:00.0000001" };
+                    yield return new object[] { new TimeSpan(long.MaxValue), constFormat, info, "10675199.02:48:05.4775807" };
```
Comment (danmosemsft) [inline src/System.Runtime/tests/System/TimeSpanTests.cs]: Nit instead of TimeSpan(long.MaxValue) you could write TimeSpan.MaxValue, and same for Min.

**Comment by stephentoub** [comment]
> although it would need more careful analysis to convince myself this is completely comprehensive

It's a good thing then we're not prohibiting PRs after this :)

File: src/System.Runtime/tests/System/TimeSpanTests.cs (hunk 1 above)
Comment (stephentoub) [reply src/System.Runtime/tests/System/TimeSpanTests.cs]: Added.

File: src/System.Runtime/tests/System/TimeSpanTests.cs (hunk 2 above)
Comment (stephentoub) [reply src/System.Runtime/tests/System/TimeSpanTests.cs]: Changed.

**Pull request merged**